flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStoreTest.java

```diff
@@ -232,12 +232,13 @@ public long getStateSize() {
      * (i.e., there exists an exception thrown by the method).
      */
     @Test
-    public void testAddCheckpointWithFailedRemove() throws Exception {
+    void testAddCheckpointWithFailedRemove() throws Exception {
 
         final int numCheckpointsToRetain = 1;
         final Configuration configuration = new Configuration();
         configuration.setString(
-                HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM, zooKeeperResource.getConnectString());
+                HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM,
+                zooKeeperExtensionWrapper.getCustomExtension().getConnectString());
 
         final CuratorFrameworkWithUnhandledErrorListener curatorFrameworkWrapper =
                 ZooKeeperUtils.startCuratorFramework(configuration, NoOpFatalErrorHandler.INSTANCE);
```

Comment:
it seems here `ZooKeeperUtils.startCuratorFramework` opens resources and doesn't close it.. May be put it in `try` resources block